tpp-app/src/home/pages/CalendarScreen.js

```diff
@@ -81,20 +74,73 @@ export const Calendar = ({navigation}) => {
                 },
             }
         }}
+
+        markedDates={marked} 
         />
     )
 }
 
-
 // Calendar Screen component that can be accessed by other functions
 export default function CalendarScreen ({ navigation }) {
     const [dropdownExpanded, setDropdownExpanded] = useState(false);
     const [selectedView, setSelectedView] = useState(VIEWS.Nothing);
+    const [yearInView, setYearInView] = useState([new Date().getFullYear()])
+
+    // Generate dummy initial year that just creates a blank symptom object for the year
+    const initialYear = {}
+    initialYear['0000'] = [new Symptoms()]
+    const [yearData, setYearData] = useState(initialYear)
+
+    const initialMarked = {};
+    initialMarked[new Date().toISOString().substring(0,10)] = {
+        symptoms: new Symptoms(),
+    }
+    const [marked, setMarked] = useState(initialMarked)
+
+    useEffect(() => {
+        // Whenever the user scrolls and changes what year is in view
+        yearInView.forEach(year => {
+            let yearNumber = year.toString()
+            // If the data for that year doesn't already exist
+            if (yearData[yearNumber] === undefined) {
+                newData = GETYearData(year)
+
+                const newYear = {...yearData, ...newData};
+                setYearData(newYear)
+
+                newMarkedData = {}
+                // We know that this data is now in the variable, so now attempt
+                // to convert it into the appropriate key and value data
+                let monthArray = newYear[yearNumber]
+                if (monthArray) {
+                    for (var i = 0; i < monthArray.length; i++) {
+                        for (var j = 0; j < monthArray[i].length; j++) {
+                            let date = new Date(year, i, j + 1)
+                            let isoDate = date.toISOString().substring(0,10);
+                            let symptomData = monthArray[i][j]
+    
+                            // Add it into the marked state, which then updates the calendar
+                            newMarkedData[isoDate] = {
+                                symptoms: symptomData
+                            }
+                        }
+                    }
+    
+                }
+                setMarked(markedState => ({...markedState, ...newMarkedData}))
```

Comment:
how does this work? Aren't u setting `Marked` to a callback function? I'm assuming this is just JS syntax i don't know

Reply:
@kenneth-miura It's the [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax#spread_in_object_literals). Handy for combining stuff in one line. 
This basically combines `markedState` and `newMarkedData` into one object and then returns it to be set in `setMarked`.

Reply:
yoooo that's lit